lib/license_scout/dependency_manager/gomod.rb

```diff
@@ -0,0 +1,71 @@
+#
+# Copyright:: Copyright 2020, Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'license_scout/dependency_manager/base'
+
+module LicenseScout
+  module DependencyManager
+    class Gomod < Base
+
+      def name
+        'golang_modules'
+      end
+
+      def type
+        'golang'
+      end
+
+      def signature
+        'go.sum file'
+      end
+
+      def install_command
+        'go mod download'
+      end
+
+      def detected?
+        File.exist?(go_sum_file)
+      end
+
+      def dependencies
+        go_modules.map do |mod|
+          next if mod['Main'] == true
+
+          dep_name = mod['Path']
+          dep_version = mod['Version']
+          dep_path = mod['Dir']
+
+          new_dependency(dep_name, dep_version, dep_path)
+        end.compact
+      end
+
+      def go_sum_file
+        File.join(directory, 'go.sum')
+      end
+
+      def go_modules
+        FFI_Yajl::Parser.parse(Dir.chdir(directory) {
+          go_modules_json
+        })
+      end
+
+      def go_modules_json
+        '[' + `go list -m -json all`.gsub("}\n{", "},\n{") + ']'
```

Comment:
Rather than use backticks, can you please use `Mixlib::Shellout`? That way we can make sure to properly handle exceptions.

Reply:
Can you link to an example in the code base?

Reply:
Here is an example: https://github.com/chef/license_scout/blob/master/lib/license_scout/dependency_manager/mix.rb#L73-L87

You won't need to have the native parser layer. Instead, you would pass `go list -m -json all` into `Mixlib::Shellout`, catch any exceptions, and then parse the STDOUT.